metricflow/sql_clients/databricks.py

```diff
@@ -142,6 +142,12 @@ def params_or_none(bind_params: SqlBindParameters) -> Optional[Dict[str, SqlColu
         """If there are no parameters, use None to prevent collision with `%` wildcard."""
         return None if bind_params == SqlBindParameters() else bind_params.param_dict
 
+    def _execute_stmt(
```

Comment:
The other option is to have a _log_before_execute or similar and then call that inside execute, but of course that only works if we just need to make a logging call before we run the query.

Might be worth adding a docstring to explain why this is set up this way in a general manner (so people can add custom logging hooks before and after the call to execute, or whatever description is appropriate).

Reply:
We're looking to add a Datadog trace, for which the syntax is
```
with tracer.trace:
    <SQL execution here>
```
So I think we do unfortunately have to override the whole thing! But then I can just call `super()._execute_stmt()` so hopefully not too much redundancy here.
For all other SQL clients, we use SQLAlchemy which has a nice integration with Datadog automatically, so this trace is much more easily created. (Something to keep in mind in the future when we're deciding whether or not to use SQLAlchemy for other DB engines.)

I will add a docstring!